include/dd/Package.hpp

```diff
@@ -2153,6 +2178,63 @@ namespace dd {
             return r;
         }
 
+        std::map<std::string, double> getProbVectorFromDensityMatrix(dEdge e, double measurementThreshold) {
+            std::map<std::string, double> measuredResult = {};
+            double                        p0, p1;
+            double                        globalProbability;
+            double                        statesToMeasure;
+
+            dEdge::alignDensityEdge(&e);
+
+            statesToMeasure = std::pow(2, nqubits);
+
+            for (int m = 0; m < statesToMeasure; m++) {
+                int currentResult        = m;
+                globalProbability        = dd::CTEntry::val(e.w.r);
+                std::string resultString = intToString(m, '1');
+                dEdge       cur          = e;
+                for (size_t i = 0; i < nqubits; ++i) {
+                    if (cur.p->v != -1 && globalProbability > measurementThreshold) {
+                        assert(dd::CTEntry::approximatelyZero(cur.p->e.at(0).w.i) && dd::CTEntry::approximatelyZero(cur.p->e.at(3).w.i));
+                        p0 = dd::CTEntry::val(cur.p->e.at(0).w.r);
+                        p1 = dd::CTEntry::val(cur.p->e.at(3).w.r);
+                    } else {
+                        globalProbability = 0;
+                        break;
+                    }
+
+                    if (currentResult % 2 == 0) {
+                        cur = cur.p->e.at(0);
+                        globalProbability *= p0;
+                    } else {
+                        cur = cur.p->e.at(3);
+                        globalProbability *= p1;
+                    }
+                    currentResult = currentResult >> 1;
+                }
+                if (globalProbability > 0) { // No need to track probabilities of 0
+                    measuredResult.insert({resultString, globalProbability});
+                }
+            }
+            return measuredResult;
+        }
+
+        [[nodiscard]] std::string intToString(long targetNumber, char value) const {
+            if (targetNumber < 0) {
+                throw std::runtime_error(std::string{"Integer must be at least zero!"});
+            }
+            auto        qubits = nqubits;
+            std::string path(qubits, '0');
+            auto        number = (unsigned long)targetNumber;
```

Comment:
please avoid C-style casts and use appropriate C++-style casts. In this case `static_cast`.

Reply:
I removed the cast